samtranslator/model/apigatewayv2.py

```diff
@@ -57,39 +58,160 @@ class ApiGatewayV2ApiMapping(Resource):
-            raise InvalidResourceException(api_logical_id, "AuthorizationScopes must be a list.")
-
-        # Currently only one type of auth
-        self.auth_type = "oauth2"
-
         self.api_logical_id = api_logical_id
         self.name = name
         self.authorization_scopes = authorization_scopes
-
-        # Validate necessary parameters exist
-        if not jwt_configuration:
-            raise InvalidResourceException(api_logical_id, name + " Authorizer must define 'JwtConfiguration'.")
         self.jwt_configuration = jwt_configuration
-        if not id_source:
-            raise InvalidResourceException(api_logical_id, name + " Authorizer must define 'IdentitySource'.")
         self.id_source = id_source
+        self.function_arn = function_arn
+        self.function_invoke_role = function_invoke_role
+        self.identity = identity
+        self.authorizer_payload_format_version = authorizer_payload_format_version
+        self.enable_simple_responses = enable_simple_responses
+
+        # Validate parameters
+        if authorization_scopes is not None and not isinstance(authorization_scopes, list):
+            raise InvalidResourceException(api_logical_id, "AuthorizationScopes must be a list.")
+
+        # Validate necessary parameters exist
+        authorizer_type = self._get_type()
+        if authorizer_type == "JWT":
+            if not self.jwt_configuration:
+                raise InvalidResourceException(
+                    api_logical_id, name + " OAuth2 Authorizer must define 'JwtConfiguration'."
+                )
+            if not self.id_source:
+                raise InvalidResourceException(
+                    api_logical_id, name + " OAuth2 Authorizer must define 'IdentitySource'."
+                )
+
+        if authorizer_type == "REQUEST":
+            if not self.function_arn:
+                raise InvalidResourceException(api_logical_id, name + " Lambda Authorizer must define 'FunctionArn'.")
+            if not self.authorizer_payload_format_version:
+                raise InvalidResourceException(
+                    api_logical_id, name + " Lambda Authorizer must define 'AuthorizerPayloadFormatVersion'."
+                )
+
+    def _get_type(self):
+        if self.jwt_configuration:
+            return "JWT"
+        return "REQUEST"
 
     def generate_openapi(self):
         """
         Generates OAS for the securitySchemes section
         """
-        openapi = {
-            "type": self.auth_type,
-            "x-amazon-apigateway-authorizer": {
-                "jwtConfiguration": self.jwt_configuration,
-                "identitySource": self.id_source,
-                "type": "jwt",
-            },
-        }
+        authorizer_type = self._get_type()
+        APIGATEWAY_AUTHORIZER_KEY = "x-amazon-apigateway-authorizer"
+        if authorizer_type == "JWT":
+            openapi = {
+                "type": "oauth2",
+                "x-amazon-apigateway-authorizer": {
```

Comment:
minor: openapi[APIGATEWAY_AUTHORIZER_KEY] = {...} is more consistent with another branch.